src/tests/tests.cpp

```diff
@@ -760,6 +774,19 @@ std::string Test::data_file(const std::string& what)
    return Test::data_dir() + "/" + what;
    }
 
+//static
+std::string Test::data_file_as_temporary_copy(const std::string &what)
```

Comment:
#3150 will add a test that takes an SQLite file generated by the Botan 2.x `Session_Manager_SQLite` and tries to open it with the new implementation. That will trigger a database migration (rudimentary: wipe and rebuild) which writes into the database file. The 'old' file is checked into the `test-data` directory and every time the test is run, we need to create a temporary copy of that database file to work on.

I found it important to have such a test, despite the simple migration strategy. After all, users will likely wind up opening old 2.x database files in production after migrating to 3.0.

Reply:
Now, I believe, SQLite has a pluggable storage interface that would allow us to implement such a test in-memory rather than in a temp file. Though, I found the temporary file approach much easier to pull off.

Reply:
OK now I understand.

Rather than plugging sqlite's storage it seems a lot easier to create a mock `Database` subclass. That TBH seems better to me, in that it allows verifying the specific expected sql statements are produced. But quite a bit more work than just copying and then examining the database post-migration. OK to proceed with your approach for now, I'll maybe try this mock approach later on in the year since we could use it for testing the other DB backed code.